### ISSUE TITLE
Validity range of dynamic viscosity function for ideal gases

### ISSUE THREAD

**Issue by modelica-trac-importer** [modelica/Modelica#1171]
**Reported by fcasella on 17 Jun 2013 22:12 UTC**
Consider the following test case:

```
model TestAir
  package Medium = Modelica.Media.Air.DryAirNasa;
  Medium.ThermodynamicState state;
  Medium.AbsolutePressure p = 1e5;
  Medium.Temperature T;
  Medium.SpecificHeatCapacity cp;
  Medium.DynamicViscosity mu;
  Real Pr;
equation 
  T = 300+time*3000;
  state = Medium.setState_pT(p,T);
  cp = Medium.specificHeatCapacityCp(state);
  mu = Medium.dynamicViscosity(state);
  Pr = Medium.prandtlNumber(state);
end TestAir;
```

Beyond T = 1560, the dynamic viscosity and the Prandtl number become negative. Clearly, the validity range of the function `Modelica.Media.IdealGases.Common.Functions.dynamicViscosityLowPressure()` is smaller than the interval 200-6000 K claimed in the package documentation.

I would suggest to clarify in the documentation what the validity range is for this function, and/or add an assert with the appropriate bounds inside the function body.

----

Migrated-From: https://trac.modelica.org/Modelica/ticket/1171

**Comment by modelica-trac-importer** [comment]
**Comment by otter on 18 Jun 2013 10:30 UTC**
In the documentation of DryAirNasa the following description text is present both for dynamicViscosity and thermalConductivity

  "Simple polynomial for dry air (moisture influence small), valid from 73.15 K to 373.15 K"

so, the medium range seems to be quite small as compared to the specific heat capacity that has a range 200-6000 K.

What to do? Just inserting an assert in dynamicViscosity and thermaConductivity?

**Comment by modelica-trac-importer** [comment]
**Comment by otter on 18 Jun 2013 10:40 UTC**
I quickly searched on the Web and found here:

http://www.google.de/url?sa=t&rct=j&q=dynamic%20viscosity%20of%20air%20polynomial&source=web&cd=7&cad=rja&ved=0CE8QFjAG&url=http%3A%2F%2Fwww.uaf.edu%2Facep%2Ffacilities%2Fdiesel-engine-test-bed%2Fexhaust-heat-recovery-sys%2FProgram-Manual.pdf&ei=uzfAUeOWIIrntQa-9YHQBw&usg=AFQjCNGRriFNaHM2uug-_uHpHP3K3HM4sQ&bvm=bv.47883778,d.Yms

a document where 3rd order polynomials for air are provided for dynamic viscosity and thermal conductivity in the range 200 - 1500 K

**Comment by modelica-trac-importer** [comment]
**Comment by beutlich on 18 Jun 2013 18:27 UTC**
Direct link: http://www.uaf.edu/acep/facilities/diesel-engine-test-bed/exhaust-heat-recovery-sys/Program-Manual.pdf

**Comment by modelica-trac-importer** [comment]
**Comment by otter on 27 Jun 2013 14:54 UTC**
Fixed in cfe27e5b04a13b052d607233b51da9275e207c7d:

Replaced dynamic viscosity and thermal conductivity computation by the wider range from MoistAir (which neglects influence of moisture). The range is now 123 .. 1273 K. Outside of this range linear extrapolation is used, so that the error in a much larger range up to 6000 K is not very large.

This required to introduce a new polynomial evaluate function in Modelica.Media.Incompressible.TableBased.Polynomials_Temp that supports linear extrapolation outside of the give valid range (evaluateWithRange, evaluateWithRange_der).

Furthermore, a test model was included in ModelicaTest that compares with a pure ideal gas approximation and checks whether the variables can be differentiated.

The analysis of the new implementation is available from the [attachment above](https://trac.modelica.org/Modelica/raw-attachment/ticket/1171/DryAirNasa-DynamicViscosity.pdf).
